### PR TITLE
feat(shared): remove the "async" useAsyncLocalePublishStatus

### PR DESCRIPTION
**Pull request by z0al**

### CODE REVIEW

File: packages/_shared/src/hooks/useLocalePublishStatus.ts [hunk 1]
```diff
@@ -37,15 +42,15 @@ function getLocalePublishStatusMap(entity: AssetProps | EntryProps, locales: Loc
 /**
  * Get the publish status for each locale
  */
-export function useAsyncLocalePublishStatus(
+export function useLocalePublishStatus(
```
Comment (chrishelgert) [inline packages/_shared/src/hooks/useLocalePublishStatus.ts:45]: This is also used where we don't have the localesAPI (entry list, entry editor), they would no longer work would they?

Comment (chrishelgert) [reply packages/_shared/src/hooks/useLocalePublishStatus.ts:45]: Can we support both, providing a list of Locale[] or LocaleCodes[] and the Locales API?

Comment (z0al) [reply packages/_shared/src/hooks/useLocalePublishStatus.ts:45]: Ah, so we want to keep both variants?

Comment (chrishelgert) [reply packages/_shared/src/hooks/useLocalePublishStatus.ts:45]: Yeah I think we have to yes
This are all the places where it's used in user_interface and we don't have the LocalesAPI everywhere
https://github.com/search?q=repo:contentful/user_interface%20useAsyncLocalePublishStatus&type=code

Comment (z0al) [reply packages/_shared/src/hooks/useLocalePublishStatus.ts:45]: > providing a list of Locale[] or LocaleCodes[] and the Locales API?

Why LocaleCodes[]?

Comment (z0al) [reply packages/_shared/src/hooks/useLocalePublishStatus.ts:48]: Reverted back support for `LocaleProps[]`

Comment (chrishelgert) [reply packages/_shared/src/hooks/useLocalePublishStatus.ts:48]: > Why LocaleCodes[]?
As only the `locale.code` would be needed maybe that's more a simpler then 🤷

**Review by chrishelgert — APPROVED** [review]
Thanks for taking care of this

File: packages/_shared/src/hooks/useLocalePublishStatus.ts (hunk 1 above)
Comment (z0al) [reply packages/_shared/src/hooks/useLocalePublishStatus.ts:48]: Not really, I don't know why but our status map is returned with `status` & `locale` as a body where the locale needs to at least have a `code`, `default` and `name` which is later used to render the status badge. 

I don't like it but I'm also not too familiar with where these things are used so didn't want to explode the scope. Would probably create a separate refactoring ticket, wdyt?

**Pull request merged**